src/ams.py

```diff
@@ -121,36 +121,40 @@ def __init__(self, charm: ops.CharmBase):
         self._sc = snap.SnapCache()
         self._charm = charm
 
+    @property
+    def snap(self):
+        """Return AMS snap."""
+        return self._sc[SNAP_NAME]
+
     def restart(self):
         """Restart AMS Snap."""
-        self._sc["ams"].restart()
+        self.snap.restart()
 
     def remove(self):
         """Remove AMS users, drop-in service and the snap."""
-        self._sc["ams"].remove()
+        snap.remove(SNAP_NAME)
         shutil.rmtree(SERVICE_DROP_IN_PATH.parent)
         passwd.remove_group(GROUP_NAME)
 
-    def install(self):
+    def install(self, channel: str, revision: Optional[str] = None):
         """Install AMS including its Snap."""
         try:
-            res = self._charm.model.resources.fetch("ams-snap")
-            # FIXME: Install the ams snap from a resource until we make the
-            # snaps in the snap store unlisted
-            if res is not None and res.stat().st_size:
-                snap.install_local(res, classic=False, dangerous=True)
-                logger.info("Installed AMS snap from local snap resource")
+            kwargs = {}
+            if revision:
+                kwargs = {"revision": int(revision)}
             else:
-                raise Exception("Invalid format for `ams-snap` resource")
-        except ops.ModelError:
-            self._charm.unit.status = BlockedStatus("Waiting for AMS snap resource")
-            raise
+                kwargs = {"channel": channel}
+            self.snap.ensure(state=snap.SnapState.Latest, **kwargs)
```

Comment:
This works fine, I have tested it. This is because holds are applied mainly to auto refreshes, what we do here is call refresh with a target which is manual refreshing which always works. This is also confirmed by the documentation [here](https://snapcraft.io/docs/managing-updates):
```
The refresh hold is:

Effective on auto-refreshes and general snap refresh requests
Not effective on targeted snap refreshes
```

Reply:
Okay, thanks for the confirmation.